internal/isoeditor/rhcos.go

```diff
@@ -108,21 +129,45 @@ func (e *rhcosEditor) CreateClusterMinimalISO(ignition string, staticIPConfig st
 
 func (e *rhcosEditor) embedInitrdPlaceholders() error {
 	// Create ramdisk image placeholder
-	if err := e.createImagePlaceholder(ramDiskImagePath); err != nil {
+	if err := e.createImagePlaceholder(ramDiskImagePath, RamDiskPaddingLength); err != nil {
 		return errors.Wrap(err, "Failed to create placeholder for custom ramdisk image")
 	}
 
 	return nil
 }
 
-func (e *rhcosEditor) createImagePlaceholder(imagePath string) error {
+func (e *rhcosEditor) embedOffsetsInSystemArea(isoPath string) error {
+	ignitionOffset, err := isoutil.GetFileLocation(ignitionImagePath, isoPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get ignition image offset")
+	}
+
+	ramDiskOffset, err := isoutil.GetFileLocation(ramDiskImagePath, isoPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get ram disk image offset")
+	}
+
+	var ignitionOffsetInfo OffsetInfo
+	copy(ignitionOffsetInfo.Key[:], ignitionHeaderKey)
+	ignitionOffsetInfo.Offset = ignitionOffset
+	ignitionOffsetInfo.Length = IgnitionPaddingLength
```

Comment:
Do you think it would make sense to also pull the length from the actual file in the iso rather than assuming it's the length we have here?

Not sure why there would ever be a difference, but if they do somehow end up different I would rather get both values from the iso.

Reply:
should not make a difference indeed. but for reading it from the iso, wouldn't we have to add something like Location in
go-diskfs first?:
https://github.com/diskfs/go-diskfs/blob/a9a84ddce08abe8d70b5244f56eafe53d119d0ca/filesystem/iso9660/file.go#L80
I.e.
func (fl *File) Size() uint32 {
	return fl.size
}

Reply:
https://github.com/diskfs/go-diskfs/blob/ba492710e2d846eba20431917605d262200b7b81/filesystem/iso9660/file.go#L11
and
https://github.com/diskfs/go-diskfs/blob/ba492710e2d846eba20431917605d262200b7b81/filesystem/iso9660/directoryentry.go#L476

Make `Size()` part of the API already, should just be able to call it

Reply:
ok, it's already available :) https://github.com/diskfs/go-diskfs/blob/ba492710e2d846eba20431917605d262200b7b81/filesystem/iso9660/directoryentry.go#L476